test/unit/polyfills/test-intersection-observer.js

```diff
@@ -105,12 +114,36 @@ describes.fakeWin('install', {}, (env) => {
     expect(win.IntersectionObserver).to.equal(IntersectionObserverStub);
   });
 
-  it('should keep native when available', () => {
+  it('Unsupported root:document: should return native when non-document root requested', () => {
     const {win} = env;
     const native = function () {};
     win.IntersectionObserver = native;
     install(win);
-    expect(win.IntersectionObserver).to.equal(native);
+    expect(new win.IntersectionObserver(() => {})).instanceOf(native);
+    expect(
+      new win.IntersectionObserver(() => {
+        root: null;
+      })
+    ).instanceOf(native);
+    expect(
+      new win.IntersectionObserver(() => {
+        root: {
+          nodeType: 1;
+        }
+      })
+    ).instanceOf(native);
+  });
+
+  it('should return stub when {root:document} requested', () => {
```

Comment:
Do we have a test that this specific case gets upgraded correctly?

Reply:
now we do :)